02-documentation/developer-guides/building-a-nextjs-blog-with-squidex.md

```diff
@@ -0,0 +1,585 @@
+
+```jsx
+import BlogItem from "../components/blogItem";
+```
+
+And in the `Home` function in `index.js`, after the `Head` tag, add the following:
+
+```jsx
+<div className="blog-post-list">
+  {
+    posts.map((post) => <BlogItem title={post.flatData.title} slug={post.flatData.slug} key={post.id} />)
+  }
+</div>
+```
+
+This loops through your posts and creates a link to each of them.
+
+Your completed `index.js` file will look like this:
+
+```jsx
+import Head from "next/head";
+import fetchAPI from "../lib/squidex";
+import BlogItem from "../components/blogItem";
+
+export default function Home({ posts }) {
+  return (
+    <div>
+      <Head>
+        <title>My blog</title>
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+      <div className="blog-post-list">
+        {
+          posts.map((post) => <BlogItem title={post.flatData.title} slug={post.flatData.slug} key={post.id} />)
+        }
+      </div>
+    </div>
+  );
+}
+
+export async function getStaticProps() {
+  const data = await fetchAPI(`
+  {
+    queryPostsContents {
+      id,
+      flatData {
+        title,
+        slug,
+      }
+    }
+  }
+  `)
+  return {
+    props: {
+      posts: data.queryPostsContents
+    }
+  }
+}
+```
+
+If you look at your blog homepage in a browser, you will see that the titles of your posts are shown and they link to the page where the article can be read.
+
+Now that you have successfully created links to your posts on the homepage, you will create the pages where your blog posts can be read.
+
+## Step 7 — Creating the blog post pages
+
+In the `pages` folder, create a `[slug].js` file. This kind of route in a Next.js app is known as a dynamic route. Learn more about dynamic routes and how they are used from [the Next.js docs](https://nextjs.org/docs/routing/dynamic-routes).
+
+```jsx
+export default function BlogPost() {
+    return (<div>
+        Body
+    </div>)
+}
+```
+
+Click one of the links on the homepage to visit the page where your blog posts will be read.
+
+Since you will need to render markdown content on this page, you will install [next-mdx-remote](https://github.com/hashicorp/next-mdx-remote), a package that parses markdown content and renders it in your Next.js application. Stop the development server by pressing `Ctrl/Cmd + C` on your terminal and run this command to install `next-mdx-remote`:
+
+```bash
+yarn add next-mdx-remote
+```
+
+Restart the development server after installation:
+
+```
+yarn dev
+```
+
+Import `serialize` and `MDXRemote` from `next-mdx-remote` at the top of `[slug].js`:
+
+```jsx
+import { serialize } from 'next-mdx-remote/serialize';
+import { MDXRemote } from 'next-mdx-remote';
+```
+
+Import `fetchAPI` at the top of `[slug].js` since you need to fetch your blog post from Squidex:
+
+```js
+import fetchAPI from '../lib/squidex';
+```
+
+Export a `getStaticProps` function from `[slug].js`:
+
+```js
+export async function getStaticProps({ params }) {
```

Comment:
The code formatting is broken, because you use tabs and spaces. This can lead to inconsistencies.

Reply:
I just formatted the article with prettier. Let me know if that fixes it.

Reply:
I have not tested it, but I guess som

Reply:
I still see a mix of tabs and spaces.

Reply:
I sent in a commit that fixes it. Let me know if all the indentations are good, and if it is not, which lines need to be fixed.